blueocean-pipeline-editor/src/main/js/EditorPage.jsx

```diff
@@ -103,8 +111,15 @@ class SaveDialog extends React.Component {
                     onChange={o => this.setState({branch: o.branch})} disabled={this.state.saving} />
                 <div className="indent-form" style={{marginBottom: '-6px'}}>
                 <FormElement className="customBranch" errorMessage={this.state.branchError}>
-                    <TextInput placeholder="my-new-branch" onChange={value => this.setState({branch: this.branchOptions[1].branch = value})}
-                        disabled={this.state.branch !== this.branchOptions[1].branch || this.state.saving} />
+                    <div className="TextInput">
+                        <input
+                            className="TextInput-control"
+                            type="text"
+                            value={this.state.branch}
+                            onChange={this.processBranchName}
```

Comment:
only issue is, i would do `onChange={event => this.processBranchName(event)}` instead of the `.bind(this)` earlier